server/modules/analysis/repository/chain/update.js

```diff
@@ -1,31 +1,63 @@
-import { TableChain } from '../../../../../common/model/db'
+import * as DB from '../../../../db'
 
-import { db } from '../../../../db/db'
-import * as DbUtils from '../../../../db/dbUtils'
+import * as Chain from '../../../../../common/analysis/processingChain'
+import { TableChain } from '../../../../../common/model/db'
 
 /**
  * Updates the fields passed as argument of a processing chain.
  *
  * @param {!object} params - The query parameters.
  * @param {!string} params.surveyId - The survey id.
  * @param {!string} params.chainUuid - The processing chain uuid.
- * @param {!object<string, any>} params.fields - A <key, value> object containing the fields to update.
+ * @param {object<string, any>} [params.fields={}] - A <key, value> object containing the fields to update.
  * @param {boolean} [params.dateExecuted=false] - Whether to update date executed to current time.
+ * @param {boolean} [params.dateModified=false] - Whether to update date modified to current time.
  * @param {pgPromise.IDatabase} [client=db] - The database client.
  *
  * @returns {Promise<null>} - The result promise.
  */
-export const updateChain = (params, client = db) => {
-  const { surveyId, chainUuid, fields, dateExecuted = false } = params
+export const updateChain = async (params, client = DB.client) => {
+  const { surveyId, chainUuid, fields = {}, dateExecuted = false, dateModified = false } = params
+
+  if (!surveyId || !chainUuid)
+    throw new Error(`Survey id and chain uuid are required. {surveyId:${surveyId}, chainUuid:${chainUuid}`)
+
   const table = new TableChain(surveyId)
 
+  const setFields = Object.keys(fields).map((field, i) =>
+    field === TableChain.columnSet.props
+      ? `${TableChain.columnSet.props} = ${TableChain.columnSet.props} || $${i + 2}::jsonb`
+      : `${field} = $${i + 2}`
+  )
```

Comment:
this is almost the same as the one for Calculation, can we have a generic query builder module for those queries?

Reply:
we can, any suggestions where to put it and how it should be?

Reply:
it could be under db, there can be a function like buildSetPropsQuery that returns an object like {query, values} (with named parameters maybe, it would be easier to merge the parameters later).

Reply:
good idea. let's do this in a separate issue